# Data Preprocessing — Video Game Sales

Notebook ini berisi tahapan preprocessing data berdasarkan temuan dari EDA.

| Tahap | Keterangan |
|-------|------------|
| 1 | Import Library & Load Data |
| 2 | Kondisi Data Awal |
| 3 | Menghapus Missing Value |
| 4 | Menghapus Data Duplikat |
| 5 | Mengatasi Data Tidak Konsisten |
| 6 | Mengubah Format Data |
| 7 | Feature Engineering |
| 8 | Filtering Data Tidak Relevan |
| 9 | Hasil Akhir & Export |

## 1. Import Library & Load Data

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

# Load dataset mentah
df = pd.read_csv("../data/raw/video_game_sales.csv")
print("Dataset berhasil dimuat.")
print(f"Jumlah baris : {df.shape[0]:,}")
print(f"Jumlah kolom : {df.shape[1]}")

Dataset berhasil dimuat.
Jumlah baris : 16,598
Jumlah kolom : 11


## 2. Kondisi Data Awal

Menampilkan kondisi dataset sebelum preprocessing sebagai **baseline** perbandingan.

In [4]:
print('=' * 50)
print('KONDISI DATA AWAL')
print('=' * 50)
print(f'Jumlah baris    : {df.shape[0]:,}')
print(f'Jumlah kolom    : {df.shape[1]}')
print(f'Jumlah duplikat : {df.duplicated().sum()}')
print()
print('Missing Value per Kolom:')
missing = df.isnull().sum()
missing_pct = (df.isnull().sum() / len(df) * 100).round(2)
missing_df = pd.DataFrame({'Jumlah Missing': missing, 'Persentase (%)': missing_pct})
print(missing_df[missing_df['Jumlah Missing'] > 0])
print()
print('Tipe Data:')
print(df.dtypes)

KONDISI DATA AWAL
Jumlah baris    : 16,598
Jumlah kolom    : 11
Jumlah duplikat : 0

Missing Value per Kolom:
           Jumlah Missing  Persentase (%)
Year                  271            1.63
Publisher              58            0.35

Tipe Data:
Rank              int64
Name                str
Platform            str
Year            float64
Genre               str
Publisher           str
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object


**Temuan dari EDA:**
- Kolom `Year` memiliki **271 missing value** (1.63%) — bertipe float, seharusnya integer
- Kolom `Publisher` memiliki **58 missing value** (0.35%)
- Perlu dicek adanya duplikat
- Perlu dicek konsistensi nilai `Year` (ada tahun yang tidak wajar)